extensions/content-manager/admin/src/components/SettingsViewWrapper/index.js

```diff
@@ -0,0 +1,329 @@
+
+        return (
+          ![
+            'json',
+            'text',
+            'relation',
+            'component',
+            'boolean',
+            'date',
+            'media',
+            'richtext',
+          ].includes(type) && !!type
+        );
+      });
+      return options;
+    }
+
+    if (input.name === 'settings.relationName') {
+      const options =  getRelationNameOptions()
+      return options;
+    }
+
+    return input.options;
+  };
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    toggleWarningSubmit();
+    emitEvent('willSaveContentTypeLayout');
+  };
+
+  if (isLoading) {
+    return <LoadingIndicatorPage />;
+  }
+
+  const filterInputsByRelation = () => {
+    const filteredInputs = inputs.filter(input => {
+      return input.name !== 'settings.relationKey' && input.name !== 'settings.relationName'
+    });
+    const relationNameOptions = getRelationNameOptions();
+    if(modifiedData.settings && relationNameOptions.length) {
+      return modifiedData.settings.relation
+        ? inputs 
+        : filteredInputs
+    }
+    return filteredInputs;
+  }
+
+  const generateInputValue = (input) => {
+    if(input.name === 'settings.relation'){
+      const relationNameOptions = getRelationNameOptions();
+      return relationNameOptions.length > 0 ? get(modifiedData, input.name, '') : false;
+    }
+    return get(modifiedData, input.name, '')
+  }
+
+  return (
+    <>
+      <BackHeader onClick={goBack} />
+      <Container className="container-fluid">
+        <form onSubmit={handleSubmit}>
+          <Header {...headerProps} />
+          <div
+            className="row"
+            style={{
+              paddingTop: '3px',
+            }}
+          >
+            <Block
+              style={{
+                marginBottom: '13px',
+                paddingBottom: '30px',
+                paddingTop: '24px',
+              }}
+            >
+              <SectionTitle isSettings />
+              <div className="row">
+                {filterInputsByRelation().map(input => {
+                  return (
+                    <FormattedMessage key={input.name} id={input.label.id}>
+                      {label => (
+                        <div
+                          className={input.customBootstrapClass}
+                          style={{ marginBottom: 1 }}
+                        >
+                          <FormattedMessage
+                            id={get(
+                              input,
+                              'description.id',
+                              'app.utils.defaultMessage'
+                            )}
+                          >
+                            {description => (
+                              <Input
+                                {...input}
+                                description={description}
+                                label={label === ' ' ? null : label}
```

Comment:
I am not sure what is it but probably You want to do something like `label={label.trim()}` or `label={label.trim() || null}`

Reply:
strapi code